nameko/testing/waiting.py

```diff
@@ -0,0 +1,63 @@
+""" https://github.com/mattbennett/call-waiting
+"""
+import sys
+from contextlib import contextmanager
+from threading import Semaphore
+
+import six
+from mock import patch
+
+
+class WaitResult(object):
+    res = None
```

Comment:
since `None` is a valid return value, maybe this should use a semaphore (marker, not eventlet sem) to distinguish "not done yet" from "returned `None`. 

could maybe even raise an exception if `get` is called before we're ready (this might only make sense in the context of the entrypoint_waiter, see separate comment re: if we want to limit the use of wait_for_call to that (in return for smaller api surface + things like the above)